src/ts/utils.ts

```diff
@@ -0,0 +1,9 @@
+export const $ = <E extends Element>(selector: string, baseElement: E | Document = document): E | null =>
+  baseElement.querySelector(selector);
+
+export const $$ = <E extends Element>(selector: string, baseElement: E | Document = document): NodeListOf<E> =>
+  baseElement.querySelectorAll(selector);
```

Comment:
부모요소도 받고 있기 때문에 제네릭에 인수를 하나 더 추가해 주면 좋을 것 같아요.

```suggestion
export const $ = <E extends Element, P extends Element>(
  selector: string,
  baseElement: P | Document = document,
): E | null => baseElement.querySelector(selector);

export const $$ = <E extends Element, P extends Element>(
  selector: string,
  baseElement: P | Document = document,
): NodeListOf<E> => baseElement.querySelectorAll(selector);
```

그리고 현재 대부분 클래스 필드의 타입을 정의해 주고 있고 contructor에서 클래스 필드들을 초기화 하고 있는데, 타입을 또 잡아주고 있더라구요. 삭제해도 무방할 것 같습니다. 🙂  
```ts
export default class View {
  $$tabResultContainers: NodeListOf<HTMLTableSectionElement>;
  $tabProductManageButton: HTMLInputElement;
...

  constructor(vendingMachine: VendingMachineInterface) {
    // as is
    this.$$tabResultContainers = <NodeListOf<HTMLTableSectionElement>>$$('.tab-result-container');
    this.$tabProductManageButton = <HTMLInputElement>$('#tab-product-manage');
    
    // to be
    this.$$tabResultContainers = $$('.tab-result-container');
    this.$tabProductManageButton = $('#tab-product-manage');
```

Reply:
2단계에서 피드백 반영하겠습니다 감사합니다! ㅎㅎㅎㅎ